sssd_test_framework/hosts/ad.py

```diff
@@ -94,33 +94,75 @@ def disconnect(self) -> None:
 
     def backup(self) -> None:
         """
-        Perform limited backup of the domain controller data. Currently only
-        content under ``$default_naming_context`` is backed up.
+        Perform limited backup of the domain controller data. Content under
+        ``$default_naming_context`` is backed up and list of GPOs.
 
         This is done by performing simple LDAP search on the base dn. This
         operation is usually very fast.
         """
         self.ssh.run(
             rf"""
-                $backup = "C:\multihost_backup.txt"
-                if (Test-Path $backup) {{
-                    Remove-Item $backup
-                }}
-                $result = Get-ADObject -SearchBase '{self.naming_context}' -Filter "*"
-                foreach ($r in $result) {{
-                    $r.DistinguishedName | Add-Content -Path $backup
-                }}
-            """
+        $backup = "C:\multihost_backup.txt"
+        if (Test-Path $backup) {{
+              Remove-Item $backup
+            }}
+        $result = Get-ADObject -SearchBase '{self.naming_context}' -Filter "*"
+        foreach ($r in $result) {{
+           $r.DistinguishedName | Add-Content -Path $backup
+            }}
+        """
         )
         self._backup_location = "C:\\multihost_backup.txt"
 
+        self.ssh.run(
+            rf"""
+        $backup = "C:\multihost_gpo_backup.txt"
+        if (Test-Path $backup) {{
+            Remove-Item $backup
+        }}
+        $search_base = "cn=policies,cn=system,{self.naming_context}"
+        $object_class = "groupPolicyContainer"
+        $result = Get-ADObject -SearchBase "$search_base" -Filter "ObjectClass -eq '$object_class'"
+            foreach ($r in $result) {{
+            $r.ObjectGUID | Add-Content -Path $backup
+            }}
+        """
```

Comment:
Please, indent it correctly.

black does not indent multiline scripts nicely as it must keep the blank lines intact. But we want nice indentation.